t/t7519-status-fsmonitor.sh

```diff
@@ -164,6 +166,8 @@ EOF
 
 # test that newly added files are marked valid
 test_expect_success 'newly added files are marked valid' '
+	write_script .git/hooks/fsmonitor-test<<-\EOF &&
+	EOF
```

Comment:
I guess I do not understand this diff hunk. previously, we let the test fsmonitor tell Git to look at the new files, but now we don't?

Is this a diff hunk that is not exactly necessary for the test case to pass, but merely to make the test more accurate, by _disallowing_ fsmonitor to trigger `lstat()`s on the three new files and instead forcing it to rely on the implicit information given by the git add` commands?

Reply:
Thanks for pointing this out.  The issue is that with these changes the fsmonitor data is refreshed when any command reads the index. Left unchanged the `git ls-files` would refresh and be using the dirty files from the previous test which would include the newly added files and they would be marked as dirty which would fail this test. Perhaps I can use the `test-tool dump-fsmonitor` to get the bitmap to compare against instead of using `ls-files`.

Reply:
Follow up question here: what behavior is this test specifically trying to validate?  

The comment says:

`test that newly added files are marked valid`

But based on the change you made it seems like newly added files will not always be marked as valid (i.e. if they were dirty before they will still be considered dirty after the `git add`).

Is there there something I'm missing (e.g. was the test `.git/hooks/fsmonitor-test` behaving incorrectly, and that's why it needs to be set to empty at the start of the test)?

Reply:
Follow up question here: what behavior is this test specifically trying to validate?  

The comment says:

`test that newly added files are marked valid`

But based on the change you made it seems like newly added files will not always be marked as valid (i.e. if they were dirty before they will still be considered dirty after the `git add`).

Is there there something I'm missing (e.g. was the test `.git/hooks/fsmonitor-test` behaving incorrectly, and that's why it needs to be set to empty at the start of the test)?

Reply:
Follow up question here: what behavior is this test specifically trying to validate?  

The comment says:

`test that newly added files are marked valid`

But based on the change you made it seems like newly added files will not always be marked as valid (i.e. if they were dirty before they will still be considered dirty after the `git add`).

Is there there something I'm missing (e.g. was the test `.git/hooks/fsmonitor-test` behaving incorrectly, and that's why it needs to be set to empty at the start of the test)?

Reply:
Maybe for this test the 
```
write_script .git/hooks/fsmonitor-test<<-\EOF &&
EOF
```

needs to be before the `git ls-files` so that the `git add` runs with the paths in `.git/hooks/fsmonitor-test` but for each `git add` the single path for the added file would need to be in `.git/hooks/fsmonitor-test` otherwise the last `git add` would mark the other paths as dirty and save that index out.

I'm taking a closer look at the tests because before the content of the `.git/hooks/fsmonitor-test` did not affect commands that were not refreshing cache entries whereas now any commands that reads the index will get the entries marked dirty that are in the `.git/hooks/fsmonitor-test`.  This means that even `git ls-files` which is being used to validate the CE_FSMONITOR_VALID could have that affected by what is in `.git/hooks/fsmonitor-test`.  So if left with all the paths in `.git/hooks/fsmonitor-test`, those paths will be dirty for every git command that reads the index.

So in most cases we need to make sure `.git/hooks/fsmonitor-test` is empty for any validating commands like `git ls-files`  so that the contents of that file will not change what cache entries are dirty.

I might also try using the `test-dump-fsmonitor` since that it only dumping the index entries before applying the bitmap or the `.git/hooks/fsmonitor-test` paths.

Reply:
> or each git add the single path for the added file would need to be in .git/hooks/fsmonitor-test otherwise the last git add would mark the other paths as dirty and save that index out.

Ahh, this is the part I was missing, thanks!